### ISSUE TITLE
Is that possible to know the duration of each char or word in this model's output?

### ISSUE THREAD

**Issue by hyzhan** [SeanNaren/deepspeech.pytorch#135]
I want to know if is possible to know the duration of each char or word in this model's output?
or the alignment of each char or word.

**Comment by SeanNaren** [comment]
Alignment is doable from the raw output of the model, however this will take a little bit of hacking to do so.

If this is urgent, what you can do is take the raw output from the model, find the start and end for each character in the output (by doing a BestPathDecoding) then do like `start/length * duration` to find a rough estimate on the location!

**Comment by hyzhan** [comment]
@SeanNaren Thank you for your quick reply. I had printed raw output before removing duplication. I try to use the length of raw output and the real duration, which is calculated by sox tool, to estimate the duration of each label. And it seem to show that every label lasts approximately 12ms. Can this be true?And does this relate to the window_size in model.py ?
By the way, I don't know how to realize an BestPath Decoder. Would you like to give me some papers or codes ?

**Comment by mcernak** [comment]
Hi,
First layer of the model perform 2D convolution, and by default the time dimension uses a stride of 2. So default raw output has about 2 times less frames. If you modify the model in this way:
```
        self.conv = nn.Sequential(
            nn.Conv2d(1, 32, kernel_size=(41, 11), stride=(2, 1)),
            nn.BatchNorm2d(32),
            nn.Hardtanh(0, 20, inplace=True),
            nn.Conv2d(32, 32, kernel_size=(21, 11), stride=(2, 1)),
            nn.BatchNorm2d(32),
            nn.Hardtanh(0, 20, inplace=True)
        )
```
you will get almost the same number of the frames in the raw output as with the sox (some starting and ending frames are discarded, I guess). However, the training will be approximately 2 times slower, as all speech frames are used.

**Comment by SeanNaren** [comment]
@mcernak is right, but I'd suggest taking it for granted that the output of the model is half the timesteps of the input (works really well internally for us)!

**Comment by mengya22** [comment]
As we kown,the ctc_decoder  can return a sequence that automatically eliminates the blank label and merge repeat labels.

Now, I want to get a message that contains complete greedy path information and probability before the elimination of the blank label and the merge of repeat labels. How can I get this information? Does the raw output means decode sequence contained blank label and duplication label? Is it possible to get the  probability of each label from raw output?

**Comment by SeanNaren** [comment]
Offsets have been added in #164

**Comment by ankitmundada** [comment]
@mcernak If you can still remember about this project:
Are you saying half number of frames because of the size of the next convolutional layer (Which is calculated as `[1+(W-F+2P)/S], W-width, F-Filter size, P-Padding, S-Stride`)?
Even in that case, the first convolutional layer is capturing all the input frames, right?
